src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -301,6 +301,21 @@ void PreReleaseRequest<I>::send_unlock() {
 
 template <typename I>
 void PreReleaseRequest<I>::finish() {
+  {
+    m_journal = nullptr;
+    m_object_map = nullptr;
+    RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
+    std::swap(m_object_map, m_image_ctx.object_map);
+    std::swap(m_journal, m_image_ctx.journal);
+  }
+  if (m_journal != nullptr) {
+    delete m_journal;
```

Comment:
should call send_close_journal() first?

Reply:
I think so. Objects like this should try to go through all cleanup steps even some steps are failed.